src/internal/packager2/inspect.go

```diff
@@ -0,0 +1,144 @@
+}
+
+// PackageInspectManifests inspects the manifests and charts within each component to find any container images
+func PackageInspectManifests(ctx context.Context, pkgLayout *layout2.PackageLayout, opts InspectManifestsOptions) (PackageInspectManifestResults, error) {
+	state, err := types.DefaultZarfState()
+	if err != nil {
+		return PackageInspectManifestResults{}, err
+	}
+	variableConfig := template.GetZarfVariableConfig(ctx)
+	variableConfig.SetConstants(pkgLayout.Pkg.Constants)
+	variableConfig.PopulateVariables(pkgLayout.Pkg.Variables, opts.SetVariables)
+	tmpPackagePath, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return PackageInspectManifestResults{}, err
+	}
+	defer os.RemoveAll(tmpPackagePath)
+
+	var resources []Resource
+	for _, component := range pkgLayout.Pkg.Components {
+		if len(component.Charts)+len(component.Manifests) < 1 {
+			continue
+		}
+		tmpComponentPath := filepath.Join(tmpPackagePath, component.Name)
+		err := os.MkdirAll(tmpComponentPath, helpers.ReadWriteExecuteUser)
+		if err != nil {
+			return PackageInspectManifestResults{}, err
+		}
+
+		applicationTemplates, err := template.GetZarfTemplates(ctx, component.Name, state)
+		if err != nil {
+			return PackageInspectManifestResults{}, err
+		}
+		variableConfig.SetApplicationTemplates(applicationTemplates)
+
+		if len(component.Charts) > 0 {
+			chartDir, err := pkgLayout.GetComponentDir(tmpComponentPath, component.Name, layout2.ChartsComponentDir)
+			if err != nil {
+				return PackageInspectManifestResults{}, err
+			}
+			valuesDir, err := pkgLayout.GetComponentDir(tmpComponentPath, component.Name, layout2.ValuesComponentDir)
+			if err != nil && !errors.Is(err, os.ErrNotExist) {
+				return PackageInspectManifestResults{}, fmt.Errorf("failed to get values: %w", err)
+			}
+
+			for _, chart := range component.Charts {
+				chartOverrides := make(map[string]any)
+				for _, variable := range chart.Variables {
+					if setVar, ok := variableConfig.GetSetVariable(variable.Name); ok && setVar != nil {
+						// Use the variable's path as a key to ensure unique entries for variables with the same name but different paths.
+						if err := helpers.MergePathAndValueIntoMap(chartOverrides, variable.Path, setVar.Value); err != nil {
+							return PackageInspectManifestResults{}, fmt.Errorf("unable to merge path and value into map: %w", err)
+						}
+					}
+				}
+				helmChart, values, err := helm.LoadChartData(chart, chartDir, valuesDir, chartOverrides)
+				if err != nil {
+					return PackageInspectManifestResults{}, fmt.Errorf("failed to load chart data: %w", err)
+				}
+				chartTemplate, err := helm.TemplateChart(ctx, chart, helmChart, values, opts.KubeVersion, variableConfig)
+				if err != nil {
+					return PackageInspectManifestResults{}, fmt.Errorf("could not render the Helm template for chart %s: %w", chart.Name, err)
+				}
+				resources = append(resources, Resource{
+					Content:      chartTemplate,
```

Comment:
In my testing - I am seeing the `---` separator on helm charts missing a new line and therefor being merged with the content of the last resource of the last line. 
```suggestion
					Content:      fmt.Sprintf("%s\n", chartTemplate),,
```

example: 
```
        env:
        - name: LONGHORN_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
      restartPolicy: Never
      serviceAccountName: longhorn-service-account---
```

Reply:
Great catch!